File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const set = new Set([...tangerine]);
    const arr = [...set]; // 해당 크기의 귤 개수
    console.log(arr);
    
    const sortArr = [];
    
    arr.forEach(n => sortArr.push(getCount(tangerine, n)));
    
    let tans = 0;
    let count = 0;
    
    sortArr.sort((a,b) => a-b);
    
    while(tans < k) {
        tans += sortArr[sortArr.length - 1];
        sortArr.pop();
        count++;
    }
    
    return count;
}


const getCount = (a, n) => {
    let count = 0;
    
    for(let i = 0; i < a.length; i++) {
        if(a[i] === n) count++;
    }
    
    return count;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    d = {}
    
    for i in range(len(tangerine)):
        s = tangerine[i]
        
        if s in d:
            prev = d[s]
            d[s] = prev + 1
        else:
            d[s] = 1
            
    counts = []
    for key in d:
        count = d[key]
        
        counts.append(count)

    sorted_counts = sorted(counts)
    temp = k
    res = 0
    while True:
        if len(sorted_counts) == 0:
            return res
        
        if temp <= 0:
            return res
        
        count = sorted_counts[-1]
        temp -= count
        res += 1
        sorted_counts.pop()
        
        
        
